spring-cloud-kubernetes-fabric8-discovery/src/main/java/org/springframework/cloud/kubernetes/fabric8/discovery/reactive/KubernetesReactiveDiscoveryClientAutoConfiguration.java

```diff
@@ -73,6 +81,18 @@ public KubernetesReactiveDiscoveryClient kubernetesReactiveDiscoveryClient(Kuber
 		return new KubernetesReactiveDiscoveryClient(client, properties, kubernetesClientServicesFunction);
 	}
 
+	/**
+	 * Post an event so that health indicator is initialized.
+	 */
+	@Bean
```

Comment:
to be honest, this is the only "real" change here, everything else are just tests to make sure that we work exactly the same as the k8s-client health wise in the discovery implementation.